aspnet/fundamentals/hosting.rst

```diff
@@ -0,0 +1,226 @@
+
+.. image:: hosting/_static/detailed-error-page.png
+
+Environment ``string``
+    Key: ``environment``. Defaults to "Production". May be set to any value. Framework-defined values include "Development", "Staging", and "Production". Values are not case sensitive. See :doc:`environments`. Set using the ``UseEnvironment`` method.
+
+.. code-block:: c#
+
+    .UseEnvironment("Development")
+
+.. note:: By default, the environment is read from the ``ASPNETCORE_ENVIRONMENT`` environment variable. When using Visual Studio, the environment will be based on settings in the *launchSettings.json* file.
+
+Server URLs ``string``
+    Key: ``urls``. Set to a semicolon (;) separated list of URL prefixes to which the server should respond. For example, "http://localhost:123". The domain/host name can be replaced with "*" to indicate the server should listen to requests on any IP address or host using the specified port and protocol (for example, "http://*:5000" or "https://*:5001"). The protocol ("http://" or "https://") must be included with each URL.
+
+.. code-block:: c#
+
+    .UseUrls("http://*:5000;http://localhost:5001;https://hostname:5002")
+
+Startup Assembly ``string``
+    Key: ``startupAssembly``. Determines the assembly to search for the ``Startup`` class. Set using the ``UseStartup`` method. Note this setting is overridden by ``WebHostBuilder.UseStartup<StartupType>``.
+
+.. code-block:: c#
+
+    .UseStartup("StartupAssemblyName")
+
+
+.. _web-root-setting:
+
+Web Root ``string``
+    Key: ``webroot``. If not specified, Web Root will use the Content Root path. Determines default root location for static files to be served by the app (relative to Content Root). Set using ``UseWebRoot``.
+
+.. code-block:: c#
+
+    .UseWebRoot("public")
+
+Use :doc:`configuration` to set configuration values to be used by the host. These values may be subsequently overridden. This is specified using ``UseConfiguration``.
+
+.. code-block:: c#
+  :emphasize-lines: 3-6,9
+
+  public static void Main(string[] args)
+  {
+    var config = new ConfigurationBuilder()
+      .AddCommandLine(args)
+      .AddJsonFile("hosting.json", optional: true)
+      .Build();
+
+    var host = new WebHostBuilder()
+      .UseConfiguration(config)
+      .UseKestrel()
+      .Configure(app => 
+      {
+        app.Run(async (context) => await context.Response.WriteAsync("Hi!"));
+      })
+    .Build();
+
+    host.Run();
+  }
+
+In the example above, command line arguments may be passed in to configure the host, or configuration settings may optionally be specified in a *hosting.json* file. To specify the host run on a particular URL, you could pass in the desired value from the command line:
+
+.. code-block:: none
+
+  dotnet run --urls "http://*:5000"
+
+The ``Run`` method starts the web app and blocks the calling thread until the host is shutdown.
```

Comment:
See our code at https://github.com/aspnet/Hosting/blob/dev/src/Microsoft.AspNetCore.Hosting/WebHostExtensions.cs#L13-L42 for illustration on how this is used.

This is more of an advanced scenario and I am okay with it not being in this doc. Also, the xml docs is pretty clear with how it works so I'm not too concerned.